libshaderc/include/shaderc.h

```diff
@@ -24,12 +24,26 @@ extern "C" {
 #include <stdint.h>
 
 typedef enum {
-  shaderc_glsl_vertex_shader,
-  shaderc_glsl_fragment_shader,
-  shaderc_glsl_compute_shader,
-  shaderc_glsl_geometry_shader,
-  shaderc_glsl_tess_control_shader,
-  shaderc_glsl_tess_evaluation_shader
+  // Forced shader kinds. These shader kinds force the compiler to compile the
+  // source code as the specified kind of shader.
+  shaderc_glsl_forced_vertex_shader,
+  shaderc_glsl_forced_fragment_shader,
+  shaderc_glsl_forced_compute_shader,
+  shaderc_glsl_forced_geometry_shader,
+  shaderc_glsl_forced_tess_control_shader,
+  shaderc_glsl_forced_tess_evaluation_shader,
+  // Deduce the shader kind from #pragma annotation in the source code. Compiler
+  // will emit error if #pragma annotation is not found.
+  shaderc_glsl_deduce_from_pragma,
```

Comment:
I would be ok with shaderc_glsl_deduce_from_pragma (or infer_from_pragma).
If we are too vague, then the user may expect it to deduce based on things that aren't a pragma.

For example from the use of gl_VertexID one could infer that this is a vertex shader, and gl_PatchVerticesIn could point to a tessc shader, but if we say "we only infer from pragmas" then we have a reasonable expectation that even in the presence of those other stage specific bits, we only look at the pragmas.